Capturable extended i18n

racheljmakes opened the pull request:
This is ready for review. Once reviewed, we can merge, then we can test with the new SHA in EN.
~~Should we QA this in EN before we merge by pointing to this branch in the gemfile, or should we merge it in after review and point to the new SHA for QA?~~

racheljmakes committed:
Internationalize "Sign Out" and "Sign In/Sign Up" links

We're internationalizing the signin process, and we need to
internationalize the links that come from capturable and are
used in Expectnation user accounts.

Add new locale files for devise and internationalize the sign out and
sign in/up links.

https://oreillymedia.atlassian.net/browse/ACONF-6278
https://oreillymedia.atlassian.net/browse/ACONF-6261

marko07 committed:
Internationalize NoScript message

We need to be able to see the messages on the NoScript file,
translated to Chinese.

These messages are related to:

* When Javascript code is disabled on the browser
* Messaging to tell the user that the data is being
  retrieved
* Messaging to tell the user that the plugin is
  loading

https://oreillymedia.atlassian.net/browse/ACONF-6275

marko07 committed:
Internationalize loading and error messages

We need to translate the errors with the communication with
Janrain.

Those errors are related to:

* Did not receive token back from server.
* An error occurred while logging in.

https://oreillymedia.atlassian.net/browse/ACONF-6276

ruralocity commented:
In the past, we've merged to capturable-extended, then updated EN to use the latest SHA in that branch, then pushed up for QA.

racheljmakes commented:
Cool, sounds good.

ruralocity commented:
I'm done with reviewing for now. My biggest concern is relative vs. absolute keys, but it sounds like Rachel's working on that.

ruralocity commented:
Fixups look good to me.

racheljmakes commented:
Great, I'll squash this.

racheljmakes commented:
Squashed and pushed. Maybe we should wait to merge until we can see if we get translations back from Douglas overnight. (Unless Adam is waiting on QA and gets through the QA from the first story.)

ruralocity commented:
Thanks, sounds good to me.

racheljmakes committed:
Load Janrain login widget JavaScript per-locale

We want the text in the Janrain login widget to be translated, and some
text is hardcoded in the JavaScript, it doesn't come from Janrain.

Internationalize the path to janrain_login_widget, so we can point to
translations of the JavaScript. Add new files on the CDN at the correct
locations. This will pair with a change on Expectnation's dev branch to
point to the staged version of the translated login widget. Update the
README to describe the new functionality this gives us to override the
janrain_widget_url per-locale as well as per-environment.

https://oreillymedia.atlassian.net/browse/ACONF-6368
https://oreillymedia.atlassian.net/browse/ACONF-6261

racheljmakes commented:
These fixups look good, so I'll squash them down. I'll also be adding another commit on this branch to address https://oreillymedia.atlassian.net/browse/ACONF-6368.

racheljmakes commented:
This new commit is ready for review. 

I've uploaded the Chinese version of the login widget, with placeholder "ZH" translations, and the social login section removed.

```
diff <(curl -s http://cdn.oreillystatic.com/members/js/widgets/login-widget.min.js) <(curl -s http://cdn.oreillystatic.com/members/js/widgets/login-widget.zh.min.js)
```

racheljmakes commented:
Douglas sent over these translations, so I've added them in fixups.

marko07 commented:
The fixups for the translations looks good to me.

racheljmakes commented:
Great, I'll squash them down and will point EN's i18n-janrain branch to the new commit.

racheljmakes committed:
Internationalize returning user greeting

In the Janrain JavaScript, we modify the welcome message for the
returnTraditional view of the widget, and this "Hi #{name}" is
untranslated.

Internationalize and add translation.

https://oreillymedia.atlassian.net/browse/ACONF-6358
https://oreillymedia.atlassian.net/browse/ACONF-6261

ruralocity commented on config/locales/devise.en.yml:
```diff
@@ -0,0 +1,33 @@
+      enable_javascript: Please enable JavaScript to sign in.
+      description_html: |-
+        <p>Please check your browser settings for these possible blockers, and try again once you've made any necessary changes.
+        Feel free to email <a href="mailto:accounts@oreilly.com">accounts@oreilly.com</a> if you have questions.</p>
```
Thinking out loud ... what if we pass the email address in from the host app? Would that buy us anything?

racheljmakes replied:
Yeah, I think that buys us a little, because we wouldn't have to update both apps to change the email address. I'll extract it to another translation key, so it can be overridden in the host app.

ruralocity commented:
Done reviewing this one for now. Just the one question—the other issue is a formatting cleanup to do if time allows, but I wouldn't want it to hold things up.

racheljmakes committed:
Extract customer service contact email to allow override

We're not sure which contact email will be used for Chinese events, and
it would be nice if we didn't need to change devise-capturable as well
as the host application to change this email address.

Extract email into its own key, so that it can be overridden from the
host app.

https://oreillymedia.atlassian.net/browse/ACONF-6261
https://oreillymedia.atlassian.net/browse/ACONF-6275

racheljmakes commented:
This is ready for review again. I extracted the email address as @ruralocity suggested, and I confirmed I can override it from EN.

racheljmakes commented:
Merging (after checking with Aaron).
